### PR TITLE
Support new properties for topic or queue creation

### PR DESCRIPTION
**Pull request by moarychan**
# Description

Fixes https://github.com/Azure/azure-sdk-for-java/issues/35435, sync from https://github.com/Azure/azure-sdk-for-java/pull/36996

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/spring/CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,20 @@ This section includes changes in the `spring-integration-azure-servicebus` modul
 #### Bugs Fixed
 - Fix NPE in the error handler of `ServiceBusInboundChannelAdapter` when `instrumentationManager` or `instrumentationId` is null [#36930](https://github.com/Azure/azure-sdk-for-java/pull/36930).
 
+### Spring Cloud Stream Binder Service Bus
+
+#### Features Added
+- Support two topic or queue creation options (maxSizeInMegabytes, defaultMessageTimeToLive) in Service Bus channel namespace properties [#](https://github.com/Azure/azure-sdk-for-java/pull/).
```
Comment (Netyyyy) [inline sdk/spring/CHANGELOG.md:27]: lack of PR number

File: sdk/spring/CHANGELOG.md [hunk 2]
```diff
@@ -21,6 +21,20 @@ This section includes changes in the `spring-integration-azure-servicebus` modul
...
+
+#### Breaking Changes
+- Deprecated APIs `ServiceBusChannelProvisioner.validateOrCreateForConsumer`, `ServiceBusChannelProvisioner.validateOrCreateForProducer` [#](https://github.com/Azure/azure-sdk-for-java/pull/).
```
Comment (Netyyyy) [inline sdk/spring/CHANGELOG.md:30]: same

File: sdk/spring/CHANGELOG.md [hunk 3]
```diff
@@ -21,6 +21,20 @@ This section includes changes in the `spring-integration-azure-servicebus` modul
...
+
+### Spring Azure Resource Manager
+
+#### Breaking Changes
+- Deprecated APIs `ServiceBusProvisioner.provisionQueue`, `ServiceBusProvisioner.provisionTopic`, `ServiceBusProvisioner.provisionSubscription` [#](https://github.com/Azure/azure-sdk-for-java/pull/).
+- Add new methods to provision queue and topic with entity properties [#](https://github.com/Azure/azure-sdk-for-java/pull/).
```
Comment (Netyyyy) [inline sdk/spring/CHANGELOG.md:36]: same

File: sdk/spring/CHANGELOG.md [hunk 4]
```diff
@@ -21,6 +21,20 @@ This section includes changes in the `spring-integration-azure-servicebus` modul
...
+#### Features Added
+- Support two topic or queue creation options (maxSizeInMegabytes, defaultMessageTimeToLive) in Service Bus channel namespace properties [#](https://github.com/Azure/azure-sdk-for-java/pull/).
+
+#### Breaking Changes
+- Deprecated APIs `ServiceBusChannelProvisioner.validateOrCreateForConsumer`, `ServiceBusChannelProvisioner.validateOrCreateForProducer` [#](https://github.com/Azure/azure-sdk-for-java/pull/).
+
+### Spring Azure Resource Manager
+
+#### Breaking Changes
+- Deprecated APIs `ServiceBusProvisioner.provisionQueue`, `ServiceBusProvisioner.provisionTopic`, `ServiceBusProvisioner.provisionSubscription` [#](https://github.com/Azure/azure-sdk-for-java/pull/).
```
Comment (Netyyyy) [inline sdk/spring/CHANGELOG.md:35]: same

File: sdk/spring/CHANGELOG.md [hunk 5]
```diff
@@ -21,6 +21,20 @@ This section includes changes in the `spring-integration-azure-servicebus` modul
...
 
+### Spring Cloud Stream Binder Service Bus
+
+#### Features Added
+- Support two topic or queue creation options (maxSizeInMegabytes, defaultMessageTimeToLive) in Service Bus channel namespace properties [#37151](https://github.com/Azure/azure-sdk-for-java/pull/37151).
+
+#### Breaking Changes
+- Deprecated APIs `ServiceBusChannelProvisioner.validateOrCreateForConsumer`, `ServiceBusChannelProvisioner.validateOrCreateForProducer` [#37151](https://github.com/Azure/azure-sdk-for-java/pull/37151).
+
+### Spring Azure Resource Manager
```
Comment (hui1110) [inline sdk/spring/CHANGELOG.md:32]: Is it Spring Cloud Azure Resource Manager?

**Review by Netyyyy — APPROVED** [review]

File: sdk/spring/CHANGELOG.md (hunk 5 above)
Comment (moarychan) [reply sdk/spring/CHANGELOG.md:32]: The abbreviated names are used, consistent with the others, and the full names are not used.

**Pull request merged**